src/integration.test.ts

```diff
@@ -59,69 +90,87 @@ describe("Integration Test", (): void => {
     });
   });
 
-  const mockedExec = async (load: boolean, command: string): Promise<void> => {
-    const stdout = "standard output from Bash command";
-    const stderr = "error output from Bash command";
-    child_process.exec.mockImplementationOnce(
-      (_command: any, _options: any, callback: any): any => {
-        callback(null, { stdout, stderr });
-      }
+  const assertExecBashCommand = (
+    infoCallNum: number,
+    execCallNum: number,
+    command: string,
+    stdout: string
+  ): void => {
+    expect(core.info).nthCalledWith<[string]>(infoCallNum, command);
+    expect(child_process.exec).nthCalledWith<[string, ExecOptions, any]>(
+      execCallNum,
+      command,
+      EXEC_OPTIONS,
+      expect.anything()
     );
+    expect(core.info).nthCalledWith<[string]>(infoCallNum + 1, stdout);
+    expect(core.error).lastCalledWith(STDERR);
+    expect(core.setFailed).not.toHaveBeenCalled();
+  };
 
-    await (load ? docker.loadDockerImages() : docker.saveDockerImages());
-
+  const assertLoadDockerImages = (cacheHit: boolean): void => {
     expect(core.getInput).nthCalledWith<[string, InputOptions]>(1, "key", {
       required: true,
     });
-    expect(core.info).nthCalledWith<[string]>(1, command);
-    expect(child_process.exec).lastCalledWith(
-      command,
-      EXEC_OPTIONS,
-      expect.anything()
+    expect(core.setOutput).lastCalledWith(docker.CACHE_HIT, cacheHit);
+    if (cacheHit) {
+      assertExecBashCommand(1, 1, loadCommand, STDOUT);
```

Comment:
I added an assert in the `if (cacheHit)` case testing that `core.saveState` is called exactly once.  Is it also appropriate to add an assert in the `else` case (i.e. the if there is not a cache hit) that asserts that `core.saveState` was called exactly twice?

Reply:
I wouldn't bother to assert that `core.saveState` was called exactly twice personally. I can see the argument for it, but it would imply adding a whole bunch of similar asserts elsewhere, and I don't believe it's very likely to catch a regression.

Reply:
We should assert that `child_process.exec` is called exactly once on cache hit in a pre-cursor pull request as well to keep this change as narrowly focused as possible.